File: Fila/Fila Estática com Java/FilaEstatica.java
public class FilaEstatica implements Enfileiravel{
    private int ponteiroinicio;
    private int ponteirofim;
    private Object[] dados;

    public FilaEstatica(int tamanho){
        ponteiroinicio = 0;
        ponteirofim = -1;
        dados = new Object[tamanho];
    }

    public FilaEstatica(){
        this(10);
    }

    @Override
    public boolean cheia(){
        return (ponteirofim == dados.length - 1);
    }

    @Override
    public boolean vazia(){
        return (ponteiroinicio == 0);
    }

    @Override
    public void enfileirar(Object dado){
        if(!cheia()){
            ponteirofim++;
            dados[ponteirofim] = dado;
        } else {
            System.out.println("Fila cheia.");
        }    
    }

    @Override
    public Object desenfileirar(){
        Object dadoinicio = null;
        if(!vazia()){
            dadoinicio = dados[ponteiroinicio];
            ponteiroinicio++;
        } else {
            System.out.println("Fila vazia.");
        }
        return dadoinicio;
    }

    @Override
    public Object espiar(){
        Object dadoinicio = null;
        if(!vazia()){
            dadoinicio = dados[ponteiroinicio];
        } else {
            System.out.println("Fila vazia.");
        }
        return dadoinicio;
    }

    @Override
    public void imprimir(){
        if(!vazia()){
            for(int i = ponteiroinicio; ponteiroinicio <= ponteirofim; i++){
                System.out.println("[" + dados[i] + "] ");
            }
        } else {
            System.out.println("Fila vazia.");
        }
    }
}

File: Unidade I/Fila/Fila Estática com Kotlin/FilaEstatica.kt
class FilaEstatica(tamanho: Int = 10) : Enfileiravel {

    private var ponteiroInicio = 0
    private var ponteiroFim = -1
    private var dados: Array<Any?> = arrayOfNulls(tamanho)

    override fun cheia(): Boolean {
        return ponteiroFim == dados.size - 1
    }

    override fun vazia(): Boolean {
        return ponteiroInicio > ponteiroFim
    }

    override fun enfileirar(dado: Any?) {
        if (!cheia()) {
            ponteiroFim++
            dados[ponteiroFim] = dado
        } else {
            println("Fila cheia.")
        }
    }

    override fun desenfileirar(): Any? {
        var dadoInicio: Any? = null
        if (!vazia()) {
            dadoInicio = dados[ponteiroInicio]
            ponteiroInicio++
        } else {
            println("Fila vazia.")
        }
        return dadoInicio
    }

    override fun espiar(): Any? {
        return if (!vazia()) {
            dados[ponteiroInicio]
        } else {
            println("Fila vazia.")
            null
        }
    }

    override fun imprimir() {
        if (!vazia()) {
            for (i in ponteiroInicio..ponteiroFim) {
                println("[" + dados[i] + "] ")
            }
        } else {
            println("Fila vazia.")
        }
    }
}
